File: lib/features/auth/domain/user_session.dart
import 'user_role.dart';

class UserSession {
  const UserSession({
    required this.uid,
    required this.name,
    required this.email,
    required this.role,
    required this.gender,
    this.avatar,
  });

  final String uid;
  final String name;
  final String email;
  final UserRole role;
  final String gender;
  final String? avatar;

  factory UserSession.fromMap(String uid, Map<String, dynamic> map) {
    return UserSession(
      uid: uid,
      name: map['name'] as String? ?? '',
      email: map['email'] as String? ?? '',
      role: UserRole.fromString(map['role'] as String? ?? 'editor'),
      gender: UserSession.normalizeGender(map['gender'] as String?),
      avatar: map['avatar'] as String?,
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'name': name,
      'email': email,
      'role': role.toFirestore(),
      'gender': gender,
      'avatar': avatar,
    };
  }

  /// Firestore + salik rules expect `Male` or `Female`.
  static String normalizeGender(String? value) {
    final lower = (value ?? 'Male').trim().toLowerCase();
    if (lower == 'female') return 'Female';
    return 'Male';
  }
}

File: app/src/main/java/com/example/salik_management_system/auth/domain/UserSession.kt
package com.example.salik_management_system.auth.domain

data class UserSession(
    val uid: String,
    val name: String,
    val email: String,
    val role: UserRole,
    val gender: String,
    val avatar: String? = null,
) {
    fun toMap(): Map<String, Any?> = mapOf(
        "name" to name,
        "email" to email,
        "role" to role.toFirestore(),
        "gender" to gender,
        "avatar" to avatar,
    )

    companion object {
        fun fromMap(uid: String, map: Map<String, Any?>): UserSession {
            return UserSession(
                uid = uid,
                name = map["name"] as? String ?: "",
                email = map["email"] as? String ?: "",
                role = UserRole.fromString(map["role"] as? String ?: "editor"),
                gender = normalizeGender(map["gender"] as? String),
                avatar = map["avatar"] as? String,
            )
        }

        /** Firestore + salik rules expect `Male` or `Female`. */
        fun normalizeGender(value: String?): String {
            val lower = (value ?: "Male").trim().lowercase()
            return if (lower == "female") "Female" else "Male"
        }
    }
}
